lib/cretonne/src/ir/funcname.rs

```diff
@@ -6,73 +6,70 @@
 use std::fmt::{self, Write};
 use std::ascii::AsciiExt;
 
-/// The name of a function can be any UTF-8 string.
+const HEX_CHARS: &'static [u8] = b"0123456789abcdef";
+
+/// The name of a function can be any sequence of bytes.
 ///
 /// Function names are mostly a testing and debugging tool.
 /// In particular, `.cton` files use function names to identify functions.
 #[derive(Debug, Clone, PartialEq, Eq, Default)]
-pub struct FunctionName(String);
+pub struct FunctionName(Vec<u8>);
 
 impl FunctionName {
-    /// Create new function name equal to `s`.
-    pub fn new<S: Into<String>>(s: S) -> FunctionName {
-        FunctionName(s.into())
+    /// Creates a new function name from a string.
+    pub fn with_str(s: &str) -> FunctionName {
+        FunctionName::new(s.into())
     }
-}
-
-fn is_id_start(c: char) -> bool {
-    c.is_ascii() && (c == '_' || c.is_alphabetic())
-}
 
-fn is_id_continue(c: char) -> bool {
-    c.is_ascii() && (c == '_' || c.is_alphanumeric())
+    /// Creates a new function name from a sequence of bytes.
+    pub fn new(bytes: Vec<u8>) -> FunctionName {
```

Comment:
I think you should just take a `bytes: &[u8]` slice here. We don't want to force the caller to allocate a `Vec`.